### ISSUE TITLE
[Feature]: Enable sharing of individual datasources when forking an app (instead off all datasources or none)

### ISSUE THREAD

**Issue by GreenFlux** [appsmithorg/appsmith#16577]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

The template datasource share flag (for reconnect datasource modal) is app-wide. There is no option to only share credentials for certain datasources and remove others when forking. We often want to use the mock_user (or other Postgres) table with an API, SaaS connector or SMTP connection that we do not want to share.

### Why should this be worked on?

For now, we have to not-share all credentials, making it that much more difficult for a user to reconfigure the app to a runnable state after forking. We should be able to safely share the credentials for our mock databases while still removing credentials for 3rd party APIs, SaaS connectors, etc. This will improve the forking experience for our users and enable us to build more complex templates.

**Comment by Nikhil-Nandagopal** [comment]
Rather than an in-app experience, I am assuming we should just have a backend way of enabling this since this isn't so much a user need as our own internal need

**Comment by GreenFlux** [comment]
@Nikhil-Nandagopal , yes, for now this just an internal need, and only for templates. We may need it for community contributed modules though if we build a market place in the future. But I don't think we need any frontend changes for now.
